core/src/test/scala/filodb.core/memstore/PartKeyIndexRawSpec.scala

```diff
@@ -0,0 +1,926 @@
+package filodb.core.memstore
+
+import filodb.core.query.ColumnFilter
+import filodb.core.query.Filter.{Equals, EqualsRegex, NotEquals, NotEqualsRegex}
+import filodb.core.GdeltTestData.{dataset1, dataset6, dataset7, partKeyFromRecords, readers, records, uniqueReader}
+import filodb.core.binaryrecord2.{RecordBuilder, RecordSchema}
+import filodb.core.DatasetRef
+import filodb.core.metadata.{PartitionSchema, Schemas}
+import filodb.memory.format.UnsafeUtils.ZeroPointer
+import filodb.memory.format.{UnsafeUtils, UTF8Wrapper}
+import filodb.memory.format.ZeroCopyUTF8String.StringToUTF8
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.must.Matchers.{contain, not}
+import org.scalatest.matchers.should.Matchers.{convertToAnyShouldWrapper, equal}
+
+import java.io.{File, FileFilter}
+import java.nio.file.{Files, StandardOpenOption}
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration.DurationInt
+import scala.util.Random
+
+trait PartKeyIndexRawSpec {
+  this: AnyFunSpec =>
+
+  def partKeyOnHeap(partKeySchema: RecordSchema,
+                    base: Any,
+                    offset: Long): Array[Byte] = partKeySchema.asByteArray(base, offset)
+
+  protected def createNewIndex(ref: DatasetRef,
+                               schema: PartitionSchema,
+                               facetEnabledAllLabels: Boolean,
+                               facetEnabledShardKeyLabels: Boolean,
+                               shardNum: Int,
+                               retentionMillis: Long,
+                               diskLocation: Option[File] = None,
+                               lifecycleManager: Option[IndexMetadataStore] = None): PartKeyIndexRaw
+
+  // scalastyle:off method.length
+  // scalastyle:off cyclomatic.complexity
+  def commonPartKeyTests(keyIndex: PartKeyIndexRaw, partBuilder: RecordBuilder): Unit = {
```

Comment:
All of these tests are moved as is from the existing Lucene class.  The only changes where to make the asserts accept results in any order as Tantivy doesn't guarantee order in returns.

No new test logic was added in this class.